sos/plugins/juju.py

```diff
@@ -9,113 +9,201 @@
-        except ValueError:
-            return []
-
-    @ensure_service_is_running("juju-db")
-    def export_mongodb(self):
-        collections = (
-            "relations",
-            "environments",
-            "linkednetworks",
-            "system",
-            "settings",
+            fp = open(statusfilespath, "r")
+            statusinfo = json_loads(fp.read())
+            fp.close()
+        except Exception:
+            return
+        if "applications" in statusinfo:
+            for appname, appinfo in statusinfo["applications"].items():
+                # Get the application config for every application in every
+                # model
+                self.add_cmd_output(
+                    "sudo -i -u {} juju config --format yaml -m {} {}".format(
+                        username, modelname, appname
+                    ),
+                    suggest_filename="juju_config_-m_{}_{}.yaml".format(
+                        modelname, appname
+                    ),
+                )
+
+    # Get some information pertaining to each model
+    def collect_model_output(self, username, modelsfilepath):
+        modelsinfo = {}
+        try:
+            fp = open(modelsfilepath, "r")
+            modelsinfo = json_loads(fp.read())
+            fp.close()
+        except Exception:
+            return
+        if "models" in modelsinfo:
+            for model in modelsinfo["models"]:
+                if "short-name" in model:
+                    modelname = model["short-name"]
+                    # Run these commands for each model
+                    model_cmds = [
+                        "juju debug-log -n 500 --no-tail -m " + modelname,
+                        "juju model-config --format yaml -m " + modelname,
+                    ]
+                    for cmd in model_cmds:
+                        # The file names can be long and confusing from the
+                        # juju commands, clean them up and give them an
+                        # appropriate extension if possible
+                        filename = self._mangle_command(cmd)
+                        if re.match(r".*_--format_yaml.*", filename):
+                            filename = re.sub(r"_--format_yaml", "", filename)
+                            filename += ".yaml"
+                        self.add_cmd_output(
+                            "su - " + username + ' -c "' + cmd + '"',
+                            suggest_filename=filename,
+                        )
+
+                    # Get the model status as json so we can dig deeper
+                    status = self.get_cmd_output_now(
+                        "sudo -i -u {} juju status -m {} --format json".format(
+                            username, modelname
+                        ),
+                        suggest_filename="juju_status_-m_{}.json".format(
+                            modelname
+                        ),
```

Comment:
Same here. Why all this complexity? The whole point of mangled file names is that they fit the file system, are readable by machines, and reasonable friendly to human eyes. This is just making extra work for no real gain.